AE_PunktBeschriftung/include/representations/Solution.hpp

```diff
@@ -1,32 +1,58 @@
-#pragma once
+#ifndef LABEL_MAP_SOLUTION_HPP
+#define LABEL_MAP_SOLUTION_HPP
 
-#include <representations/Instance.hpp>
-#include <representations/Point.hpp>
 
-#include <map>
+#include "Instance.hpp"
 
 class Solution {
-
-private:
-
-    const Instance& instance;
-
-    std::map<int, Point::Rectangle> placements;
-
-    friend std::ostream& operator<<(std::ostream &ostream, const Solution &solution);
+protected:
+    const Instance instance;
 
 public:
-
-    Solution(const Instance& instance);
-
-    void setLabel(int idx, Point::Corner corner);
-
-    bool contains(int idx) const;
-
-    bool checkCollision(const Point& p, Point::Corner placement, int otherIdx) const;
-
-    int size() const;
-
-    bool isFeasible() const;
-
-};
\ No newline at end of file
+    /**
+     * Operator to print the solution to something. Must be called like: solution << ostream;
+     * Its the only way i can think of to require a printable solution via << operator.
+     *
+     * @param ostream the stream to be printed to
+     * @return the out stream from the input
+     */
+    virtual std::ostream &operator<<(std::ostream &ostream) = 0;
+
+    /**
+     * Constructor
+     * @param instance1
+     */
+    explicit Solution(Instance &instance1);
+
+    /**
+     * Checks if the current state of the solution is feasible.
+     * @return true if the solution is feasible, false else
+     */
+    [[nodiscard]] virtual bool isFeasible() const = 0;
+
+    /**
+     * Sets a label for a given Point.
+     *
+     * @param idx the index of the point
+     * @param corner the corner, where the label is placed at
+     */
+    virtual void setLabel(int idx, Point::Corner corner) = 0;
+
+    /**
+     * Returns a point by its index.
+     * @param idx the index of the point
+     *
+     * @return the point itself
+     */
+    [[nodiscard]] const Point& getPoint(int idx) const;
```

Comment:
Isn't this a method that belongs to the instance? The instance stores and "owns" the points, right? (Like the getPOIs method in the PG, that also doesn't exist in the solution.) I'd rather build a getter for the instance

Reply:
Yes but I thought this can be a shortcut. It just calls the getPoint method of the instance.